Watchman.Discord.UnitTests/Responses/ResponsesControllerTests.cs

```diff
@@ -0,0 +1,246 @@
+                messagesServiceFactoryMock: messagesServiceFactoryMock,
+                responsesServiceMock: responsesServiceMock);
+
+            //Act
+            await controller.RemoveResponse(command, contexts);
+
+            //Assert
+            messagesServiceFactoryMock.Verify(x => x.Create(contexts), Times.Once());
+            responsesServiceMock.Verify(x => x.GetResponseByOnEvent(It.IsAny<string>(), It.IsAny<ulong>()), Times.Once);
+            responsesServiceMock.Verify(x => x.RemoveResponse(It.IsAny<string>(), It.IsAny<ulong>()), Times.Once);
+        }
+        [Test]
+        public async Task RemoveResponse_ShouldNotRemoveResponse()
+        {
+            //Arrange
+            var contexts = this.testContextsFactory.CreateContexts(5, 1, 1);
+            var command = new RemoveResponseCommand() { OnEvent = "test" };
+
+            var messagesServiceMock = new Mock<IMessagesService>();
+            var messagesServiceFactoryMock = new Mock<IMessagesServiceFactory>();
+            messagesServiceFactoryMock.Setup(x => x.Create(It.IsAny<Contexts>()))
+                .Returns(messagesServiceMock.Object);
+            var responsesServiceMock = new Mock<IResponsesService>();
+
+            var controller = this.testControllersFactory.CreateResponsesController(
+                messagesServiceFactoryMock: messagesServiceFactoryMock,
+                responsesServiceMock: responsesServiceMock);
+
+            //Act
+            await controller.RemoveResponse(command, contexts);
+
+            //Assert
+            messagesServiceFactoryMock.Verify(x => x.Create(contexts), Times.Once);
+            responsesServiceMock.Verify(x => x.RemoveResponse(It.IsAny<string>(), It.IsAny<ulong>()), Times.Never);
+        }
+
+        [Test]
+        public async Task AddResponse_ShouldAddResponse()
+        {
+            //Arrange
+            var contexts = this.testContextsFactory.CreateContexts(15, 1, 1);
+            var command = new AddResponseCommand() { OnEvent = "test", Message = "test" };
+            var response = new DomainModel.Responses.Response("test", "test", (ulong)43, new string[] { "test" });
+            var serverResponse = new DomainModel.Responses.Response("test1", "test1", (ulong)54, new string[] { "test1" });
+
+            var messagesServiceMock = new Mock<IMessagesService>();
+            var messagesServiceFactoryMock = new Mock<IMessagesServiceFactory>();
+            messagesServiceFactoryMock.Setup(x => x.Create(It.IsAny<Contexts>()))
+                .Returns(messagesServiceMock.Object);
+            var responsesServiceMock = new Mock<IResponsesService>();
+            responsesServiceMock.Setup(x => x.GetResponseByOnEvent(It.IsAny<string>(), DomainModel.Responses.Response.DEFAULT_SERVER_ID))
+                .Returns(Task.FromResult(response));
+            responsesServiceMock.Setup(x => x.GetResponseByOnEvent(It.IsAny<string>(), It.Is<ulong>(a => a == 0)))
```

Comment:
to `a == 0` nie ma sensu, bo to jest równoznaczne z wpisaniem tutaj `DomainModel.Responses.Response.DEFAULT_SERVER_ID`

jak już to powinno być `a != 0`, czyli serwerowe 
ale zamiast zapisu z `It.Is` możesz wrzucić po prostu `contexts.Server.Id` (dla ID serwerowego zwracasz serwerowe response)

przy czym jeśli to jest test `should add response`, to powinieneś zwracać tutaj nulla, czyli tego setupa nie powinno w ogóle być